### ISSUE TITLE
MvvmLight with WindsorCastle: Conflicting Microsoft.Practices.ServiceLocation

### ISSUE THREAD

**Issue by mysteryx93** [castleproject/Windsor#490]
I installed in my .NET project: 

- MvvmLight v5.4.1.1
- Castle.Windsor v5.0.0
- CommonServiceLocator v2.0.2
- CommonServiceLocator.WindsorAdapter.Unofficial v3.3.0

Then I initialize my container like this

    public ViewModelLocator()
    {
        var container = new WindsorContainer();
        ServiceLocator.SetLocatorProvider(() => new WindsorServiceLocator(container));
        container.Register(Component.For<MainViewModel>().ImplementedBy<MainViewModel>());
    }

Which gives this error:

> The type 'ServiceLocatorImplBase' is defined in an assembly that is
> not referenced. You must add a reference to assembly
> 'Microsoft.Practices.ServiceLocation, Version=1.3.0.0,
> Culture=neutral, PublicKeyToken=31bf3856ad364e35'.

Basically, MvvmLight v5.4.1.1 expects CommonServiceLocator v2.0.2, CommonServiceLocator.WindsorAdapter.Unofficial expects CommonServiceLocator v1.3.0, and CommonServiceLocator.WindsorAdapter expects CommonServiceLocator v1.0.0

One solution is to downgrade CommonServiceLocator to v1.3.0 and it works, but then MvvmLight also gets downgraded to v5.3.0

What's the right way to do this? How do I get MvvmLight and Windsor Castle working together?

**Comment by jonorossi** [comment]
Looks like both `CommonServiceLocator.WindsorAdapter` and `CommonServiceLocator.WindsorAdapter.Unofficial` are completely unmaintained packages, #362 has a bit more history.

It looks like the CommonServiceLocator project after moving to GitHub dropped all container support, you'd have to read around but assuming they expect users to just implement their own integration it looks like it shouldn't be too many lines of code using `IServiceLocator` or `ServiceLocatorImplBase`.
